ste/mgr-JobPartTransferMgr.go

```diff
@@ -278,13 +284,22 @@ func (jptm *jobPartTransferMgr) Info() TransferInfo {
 	// We need to set the blockSize in such way that number of blocks per blob
 	// does not exceeds 50000 (max number of block per blob)
 	if blockSize == 0 {
-		blockSize = uint32(common.DefaultBlockBlobBlockSize)
-		for ; uint32(sourceSize/int64(blockSize)) > common.MaxNumberOfBlocksPerBlob; blockSize = 2 * blockSize {
+		blockSize = common.DefaultBlockBlobBlockSize
+		for ; uint32(sourceSize/blockSize) > common.MaxNumberOfBlocksPerBlob; blockSize = 2 * blockSize {
+			if blockSize > common.AzCopyBlockSizeThreshold {
+				/*
+				 * For a RAM usage of 0.5G/core, we would have 4G memory on typical 8 core device, meaning at a blockSize of 256M,
+				 * we can have 4 blocks in core, waiting for a disk or n/w operation. Any higher block size would *sort of*
+				 * serialize n/w and disk operations, and is better avoided.
+				 */
+				blockSize = blockSize / common.MaxNumberOfBlocksPerBlob
```

Comment:
>blockSize [](start = 16, length = 9)

typo: blockSize should be equal to source size divided by max number of blocks, not the block size divided by max number of blocks